test/alphabet/alphabet_test.cpp

```diff
@@ -214,6 +224,55 @@ TYPED_TEST(alphabet, concept)
     EXPECT_TRUE(alphabet_concept<TypeParam>);
 }
 
+#if SEQAN_WITH_CEREAL
+template <typename in_archive_t, typename out_archive_t, typename TypeParam>
+void do_serialisation(TypeParam const l, std::vector<TypeParam> const & vec)
+{
+    std::string filename = std::tmpnam(nullptr);
+
+    {
+        std::ofstream os(filename, std::ios::binary);
+        out_archive_t archive(os);
+        archive(l);
+        archive(vec);
+    }
+
+    {
+        TypeParam in_l{};
+        std::vector<TypeParam> in_vec;
+
+        std::ifstream is(filename, std::ios::binary);
+        in_archive_t archive(is);
+        archive(in_l);
+        archive(in_vec);
+        EXPECT_EQ(l, in_l);
+        EXPECT_EQ(vec, in_vec);
+    }
+
+    std::remove(filename.c_str());
```

Comment:
You are right :bulb: There are other problems with the current approach, but I didn't immediately find a nicer solution, because the safer solutions delete the file when it is closed, but our file needs to be opened and closed multiple times to simulate proper serialisation and de-serialisation.

Reply:
We could put that into a `unique_ptr` with a custom deleter and have an utility function that creates that for our testcases?

```cpp
std::unique_ptr<char, delctype(&std::remove)> filename(std::tmpnam(nullptr), &std::remove);
```

In http://en.cppreference.com/w/cpp/memory/unique_ptr there is also this example:
```cpp
  {
      std::unique_ptr<std::FILE, decltype(&std::fclose)> fp(std::fopen("demo.txt", "r"),
                                                            &std::fclose);
      if(fp) // fopen could have failed; in which case fp holds a null pointer
        std::cout << (char)std::fgetc(fp.get()) << '\n';
  } // fclose() called here, but only if FILE* is not a null pointer
```
Maybe we should http://en.cppreference.com/w/cpp/io/c/tmpfile instead.

Reply:
> We could put that into a unique_ptr with a custom deleter and have a utility function that creates that for our testcases?

Yes, something like that would be possible, I will check it out.

> Maybe we should use http://en.cppreference.com/w/cpp/io/c/tmpfile instead.

No, this won't work, because, as I said above:

> > the safer solutions delete the file when it is closed, but our file needs to be opened and closed multiple times to simulate proper serialisation and de-serialisation.